apps/web/src/components/tx-flow/flows/ManagerSigners/ReviewSigners.tsx

```diff
@@ -0,0 +1,101 @@
+import { Box, Stack } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import type { SafeTransaction } from '@safe-global/safe-core-sdk-types'
+import type { ReactElement } from 'react'
+
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { createMultiSendCallOnlyTx, createTx } from '@/services/tx/tx-sender'
+import { SafeTxContext } from '../../SafeTxProvider'
+import { getOwnerStructureChangeTransaction } from '@/features/recovery/services/transaction'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import ReviewTransaction from '@/components/tx/ReviewTransactionV2'
+import FieldsGrid from '@/components/tx/FieldsGrid'
+import { maybePlural } from '@safe-global/utils/utils/formatters'
+import { TxFlowContext } from '../../TxFlowProvider'
+import type { ManageSignersForm } from '.'
+import type { TxFlowContextType } from '../../TxFlowProvider'
+import type { ReviewTransactionContentProps } from '@/components/tx/ReviewTransactionV2/ReviewTransactionContent'
+
+export function ReviewSigners({ onSubmit, children }: ReviewTransactionContentProps): ReactElement {
+  const { data } = useContext<TxFlowContextType<ManageSignersForm>>(TxFlowContext)
+  const { setSafeTx, setSafeTxError } = useContext(SafeTxContext)
+  const { safe } = useSafeInfo()
+
+  useEffect(() => {
+    if (!data) {
+      return
+    }
+
+    const transactions = getOwnerStructureChangeTransaction({
+      safe,
+      newThreshold: data.threshold,
+      newOwners: data.owners.map((owner) => ({
+        value: owner.address,
+      })),
+    })
+
+    const createSafeTx = async (): Promise<SafeTransaction> => {
+      const isMultiSend = transactions.length > 1
+      return isMultiSend ? createMultiSendCallOnlyTx(transactions) : createTx(transactions[0])
+    }
+
+    createSafeTx().then(setSafeTx).catch(setSafeTxError)
+  }, [data, safe, safe.deployed, setSafeTx, setSafeTxError])
+
+  return (
+    <ReviewTransaction onSubmit={onSubmit}>
+      {/* We cannot create a ConfirmationView out of the following as it accesses params */}
+      <Stack
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          gap: 3,
+          '& .MuiGrid-container': {
+            alignItems: 'flex-start',
+          },
+        }}
+      >
+        <FieldsGrid title="Signers">
+          <Box
+            sx={{
+              display: 'flex',
+              flexDirection: 'column',
+              gap: 2,
+              fontSize: '14px',
+            }}
```

Comment:
Box and typography accept sx params directly as props, so you don't have to use an inline object.

On a more general note, the new components in this PR mix logic and presentation heavily, down to individual styles. I would suggest to extract the presentational part into separate components.

Reply:
I migrated every possible instance of `Box`/`Typography` to use "direct" props and created a confirmation view component in 2ade5e31d8ea0c32a351612ee38508849dc818eb.